lib/include/currentSource.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+// Standard includes
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+// GeNN includes
+#include "global.h"
+#include "currentSourceModels.h"
+#include "variableMode.h"
+
+//------------------------------------------------------------------------
+// CurrentSource
+//------------------------------------------------------------------------
+class CurrentSource
+{
+public:
+    CurrentSource(const std::string &name, const CurrentSourceModels::Base *currentSourceModel,
+                const std::vector<double> &params, const std::vector<NewModels::VarInit> &varInitialisers, int hostID, int deviceID) :
+        m_Name(name),
+        m_CurrentSourceModel(currentSourceModel), m_Params(params), m_VarInitialisers(varInitialisers),
+        m_VarMode(varInitialisers.size(), GENN_PREFERENCES::defaultVarMode),
+        m_HostID(hostID), m_DeviceID(deviceID)
+    {
+    }
+    CurrentSource(const CurrentSource&) = delete;
+    CurrentSource() = delete;
+
+    //------------------------------------------------------------------------
+    // Public methods
+    //------------------------------------------------------------------------
+    //! Function to enable the use zero-copied memory for a particular state variable (deprecated use NeuronGroup::setVarMode):
+     /*! May improve IO performance at the expense of kernel performance */
+    void setVarZeroCopyEnabled(const std::string &varName, bool enabled)
```

Comment:
There's no need to add these deprecated functions for new models - I plan on removing **a lot** of deprecated stuff in GeNN 4.0.0

Reply:
I found these functions in `NeuronGroup` and I wasn't sure whether leaving them out would break anything.

Reply:
No worries - no one could expect you to predict my future GeNN plans or how the undocumented MPI support works :smile: